pd/src/components/staking.rs

```diff
@@ -69,33 +93,252 @@ impl Staking {
         tracing::debug!(curr_base_rate = ?current_base_rate);
         tracing::debug!(?next_base_rate);
 
-        // Update these in the JMT:
+        // Update the base rates in the JMT:
         self.overlay
-            .set_base_rates(current_base_rate, next_base_rate)
+            .set_base_rates(current_base_rate.clone(), next_base_rate.clone())
             .await;
 
-        let _staking_token_supply = self
-            .overlay
-            .token_supply(&STAKING_TOKEN_ASSET_ID)
-            .await?
-            .expect("staking token should be known");
+        let validator_list = self.overlay.validator_list().await?;
+        for v in &validator_list {
+            let validator = self.overlay.validator(v).await?.ok_or_else(|| {
+                anyhow::anyhow!("validator had ID in validator_list but not found in JMT")
+            })?;
+            // The old epoch's "current rate" is going to become the "previous rate".
+            let prev_rate = self
+                .overlay
+                .current_validator_rate(v)
+                .await?
+                .ok_or_else(|| {
+                    anyhow::anyhow!("validator had ID in validator_list but rate not found in JMT")
+                })?;
```

Comment:
It looks like we don't use this rate at all, except for an assert later.  Could we skip loading it entirely?